### ISSUE TITLE
Issue with num worker

### ISSUE THREAD

**Issue by Kathir-66** [MLI-lab/DeepDeWedge#25]
I am trying to perform fit model in a high performance cluster. 

Each of the nodes in the cluster have around 40 CPUs. But however, when I execute the fit model command with num of workers 10 in the config file and --cpus-per-task=10 in the slurm script. I get the error 

`UserWarning: This DataLoader will create 10 worker processes in total. Our suggested max number of worker in current system is 1, which is smaller than what this DataLoader is going to create.
`

I am attaching my config file and slurm script below

<details><summary>SLURM script</summary>
<p>
`
#!/bin/bash

#SBATCH -o log.out.%j       # Standard output log
#SBATCH -e err.err.%j       # Standard error log
#SBATCH -D ./  # Set working directory

#SBATCH -J onfleet_     # Job name

#SBATCH --partition=p.cryo     # Specify the partition
#SBATCH --nodes=1              # Number of nodes
#SBATCH --ntasks-per-node=2    # Number of tasks per node
#SBATCH --cpus-per-task=20     # Number of CPUs per task

#SBATCH --gres=gpu:gtx1080:2   # Request 2 GTX 1080 GPUs
#SBATCH --gpu-bind=verbose,per_task:1  # Bind 1 GPU per task
#SBATCH --mem=50G
#SBATCH --time=48:00:00        # Time limit hh:mm:ss

export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK
export OMP_PLACES=cores

ddw fit-model --config ./config.yaml 
`

 
</p>
</details> 

<details><summary>config</summary>
<p>
`
shared:
  project_dir: "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella"
  tomo0_files: 
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/367_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/455_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/458_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/462_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/463_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/537_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/even/538_6.79Apx.mrc"

  tomo1_files: 
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/367_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/455_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/458_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/462_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/463_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/537_6.79Apx.mrc"
    - "/sbdata/EM/projects/kaezhila/software/DeepDeWedge/polytomella/odd/538_6.79Apx.mrc"

  subtomo_size: 96
  mw_angle: 50
  num_workers: 20
  gpu: 0
  seed: 42

prepare_data:
  mask_files: []

  min_nonzero_mask_fraction_in_subtomo: 0.3
  subtomo_extraction_strides: [64, 80, 80] 
  val_fraction: 0.2

fit_model:
  unet_params_dict:
    chans: 64
    num_downsample_layers: 3
    drop_prob: 0.0
  adam_params_dict: 
    lr: 0.0004
  num_epochs: 1000
  batch_size: 5
  update_subtomo_missing_wedges_every_n_epochs: 10
  check_val_every_n_epochs: 10
  save_n_models_with_lowest_val_loss: 5
  save_n_models_with_lowest_fitting_loss: 5
  save_model_every_n_epochs: 50
  logger: "csv" 

refine_tomogram:
  model_checkpoint_file: "./fitted_model.ckpt"
  subtomo_overlap: 32
  batch_size: 10

`

</p>
</details>

**Comment by SimWdm** [comment]
Dear @Kathir-66,

sorry for my late reply! 
The warning you are describing is probably not from DeepDeWedge itself but comes from somewhere else in your system (maybe SLURM). 

Can you tell if the job is using 10 CPUs? 
If so, I think the warning is not too much of an issue.

Best,
Simon

**Comment by SimWdm** [comment]
Closing this due to inactivity. Feel free to re-open the issue if necessary 😊
